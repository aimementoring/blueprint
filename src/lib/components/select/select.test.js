import React from 'react';
import renderer from 'react-test-renderer';
import Select from './select';

describe('Select', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(
        <Select
          placeholder="Select an option"
          name="select"
          onChangeFunction={value => {}}
          required
        >
          <option value="" disabled>
            Select an option
          </option>
          {['Option 1', 'Option 2', 'Option 3', 'Option 4', 'Option 5'].map(value => (
            <option key={`value-${value}`}>{value}</option>
          ))}
        </Select>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
