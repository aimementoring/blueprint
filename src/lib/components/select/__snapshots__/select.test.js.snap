// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders properly 1`] = `
<div
  className="inputWrapper "
>
  <select
    className="select input "
    name="select"
    onChange={[Function]}
    placeholder="Select an option"
    required={true}
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      Select an option
    </option>
    <option>
      Option 1
    </option>
    <option>
      Option 2
    </option>
    <option>
      Option 3
    </option>
    <option>
      Option 4
    </option>
    <option>
      Option 5
    </option>
  </select>
</div>
`;
